src/credentials_rotators/npm-token-rotation/README.md

```diff
@@ -0,0 +1,112 @@
+# NPM Token rotation
+
+This is a project that generates CDK stacks to rotate the NPM tokens. The stack is generated based on the configuration passed in `secret_config.json` file in the root of the project. The configuration should have the following shape
+
+#### Configuring the app
+
+The `secret_config.json` configuration file should be populated with the
+information about **accessing** secret values from AWS Secrets Manager. **It must not be
+used to store the secret values themselves.**
+
+## Sample `secret_config.json`
+
+```json
+{
+  "npmLoginUsernameSecret": {
+    "arn": "arn:aws:secretsmanager:us-east-2:12345678:secret:npm-username-secret-ui4F2W",
+    "secretKey": "npm_login_username"
+  },
+  "npmLoginPasswordSecret": {
+    "arn": "arn:aws:secretsmanager:us-east-2:12345678:secret:npm-password-secret-OauC99",
+    "secretKey": "npm_login_password"
+  },
+  "npmOtpSeedSecret": {
+    "arn": "arn:aws:secretsmanager:us-east-2:12345678:secret:npm-otp-secret-WUwjkx",
+    "secretKey": "npm_otp_seed"
+  },
+  "npmAccessTokenSecrets": {
+    "secrets": [
+      {
+        "arn": "arn:aws:secretsmanager:us-east-2:12345678:secret:npm_access_token_secrets_cli-ZtI1lL",
+        "secretKey": "npm_access_token",
+        "publishConfig": {
+          "type": "Context", // This could either be Context or Environment
+          "contextName": "publish", // projectName when type is Environment
+          "slug": "gh/yuth",
+          "variableName": "NPM_PUBLISH_TOKEN",
+          "circleCiToken": {
+            "arn": "arn:aws:secretsmanager:us-east-2:12345678:secret:amplify_cli_circleci_token-cawIdq",
+            "secretKey": "token"
+          }
+        },
+        "slackWebHookConfig": {
+          // optional
+          "arn": " arn:aws:secretsmanager:us-east-2:1223333:secret:slackWebHookUrl-mCFi2K",
+          "secretKey": "url"
+        }
+      },
+      {
+        "arn": "arn:aws:secretsmanager:us-east-2:12345678:secret:npm_access_token_secrets_js-ZtI1lL",
```

Comment:
if we are publishing the same secret to multiple places, its probably intuitive to accept an array of `publishConfig` to avoid duplicating other keys?

Reply:
We should think hard if we want to reuse the same token in multiple places. I think its a security smell if we are re-using the same token in multiple places.